src/backend/distributed/planner/multi_router_planner.c

```diff
@@ -2219,6 +2207,51 @@ PlanRouterQuery(Query *originalQuery,
 }
 
 
+/*
+ * CreateDummyPlacement creates a dummy placement that can be used for zero
+ * shard queries.
+ *
+ * If local execution is enabled, the placement points to the local group.
+ * Otherwise an active worker is selected using round robin policy to create the
+ * dummy placement. In case there are no active workers, NULL pointer is returned.
+ */
+static ShardPlacement *
+CreateDummyPlacement(void)
+{
+	static uint32 zeroShardQueryRoundRobin = 0;
+	ShardPlacement *dummyPlacement = CitusMakeNode(ShardPlacement);
+
+	if (!EnableLocalExecutionPlanning)
```

Comment:
Perhaps we should use the citus.task_assignment_policy here? i.e. if it's round-robin we pick a random node and otherwise we pick localhost.

Reply:
Does this mean we should remove `citus.enable_local_execution_planning` completely? @onderkalaci suggested I create a new flag that is enabled by default. I guess this GUC is no longer needed if I use `TaskAssignmentPolicy`


```c
if(TaskAssignmentPolicy == TASK_ASSIGNMENT_ROUND_ROBIN)
{ /* round-robin */ }
else
{ /* dummy placement pointing at local node */ }
```

Reply:
I'd suggest removing the GUC.

Reply:
I removed `enable_local_execution_planning` and planner will prefer local execution if `TASK_ASSIGNMENT_ROUND_ROBIN` is not in effect.

Reply:
can this be reflected in the description of the PR?